test-case/multiple-pipeline-capture.sh

```diff
@@ -87,8 +89,8 @@ func_run_pipeline_with_type()
         dlogc "${APP_LST[$1]} -D $dev -c $channel -r $rate -f $fmt ${DEV_LST[$1]} -q"
         "${APP_LST[$1]}" -D $dev -c $channel -r $rate -f $fmt "${DEV_LST[$1]}" -q &
 
-        tmp_count=$(expr $tmp_count - 1 )
-        [[ $tmp_count -le 0 ]] && return
+        : $((tmp_count--))
+        if [ "$tmp_count" -le 0 ]; then return 0; fi
```

Comment:
@marc-hb what is the preferred style `[[ CONDTION  ]] && DO_SOMETHING` or `if [ CONDTION   ]; then DO_SOMETHING;fi`
I am quite confused here.

Reply:
Maybe you have in mind "avoid negations" comments I made in other reviews? There's no negation here in either case, these "avoid negations" comments don't apply.

`A || B` and `if A; then B; fi` never  have any compatibility problem with `set -e` (#312)

`A && B` works too most of the time but sometimes it can cause issues with `set -e` in some  corner cases. This is one of these corner cases because the statement is last in the function.

Even when it works, `A && B` is "less obviously" compatible with `set -e` because the exit value of the list can be non-zero even when everything is fine.

I'm genuinely sorry `set -e` can be complicated but in shell this is really the best we get.

Reply:
@marc-hb I think that was only some style preference here. But I do see issue here, the `set -e` will exit the script when `tmp_count` for example is 0, right?

Reply:
Yes `expr 0` was returning a failure. `expr` is bad (also for other reasons), shellcheck warns about it.

`:` always succeeds.

Reply:
> Even when it works, A && B is "less obviously" compatible with set -e because the exit value of the list can be non-zero even when everything is fine.

Explanation at http://redsymbol.net/articles/unofficial-bash-strict-mode/#short-circuiting if you're interested